extmod/modbluetooth.c

```diff
@@ -719,7 +713,10 @@ STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(bluetooth_ble_gatts_set_buffer_obj, 3
 STATIC mp_obj_t bluetooth_ble_gattc_discover_services(size_t n_args, const mp_obj_t *args) {
     mp_int_t conn_handle = mp_obj_get_int(args[1]);
     mp_obj_bluetooth_uuid_t *uuid = NULL;
-    if (n_args == 3) {
+    if (n_args == 3 && args[2] != mp_const_none) {
+        if (!mp_obj_is_type(args[2], &bluetooth_uuid_type)) {
+            mp_raise_ValueError(MP_ERROR_TEXT("uuid"));
```

Comment:
`mp_raise_TypeError` (and below)

Reply:
Done. This means using UUID rather than uuid makes sense to, so I've reverted all the other changes.